lib/api/CAnomalyJobConfig.cc

```diff
@@ -448,16 +447,48 @@ void CAnomalyJobConfig::CAnalysisConfig::parse(const rapidjson::Value& analysisC
     // The Job config has already been validated by Java before being passed to
     // the C++ backend. So we can safely assume that the detector config is a
     // non-null array - hence this check isn't strictly necessary.
+    this->seenField(m_SummaryCountFieldName);
+    this->seenField(m_CategorizationFieldName);
     if (detectorsConfig != nullptr && detectorsConfig->IsArray()) {
         m_Detectors.resize(detectorsConfig->Size());
+        int detectorIndex{0};
```

Comment:
```suggestion
        int fallbackDetectorIndex{0};
```

This suggestion relates to the comment about not introducing a very subtle dependency on the Java implementation of detector index.